### PR TITLE
WOR-452 cross service tracing from rawls

### PR DESCRIPTION
**Pull request by dvoet**
https://broadworkbench.atlassian.net/browse/WOR-452

This is a PR full of lots of little, similar changes. The gist is a switch from passing around `UserInfo` to passing around `RawlsRequestContext` which includes `UserInfo` and tracing information and maybe other things in the future. Changes were made in the places where `Span`s were explicitly created to create derived contexts instead (this also took care of some NPEs in waiting).

The point of all this is reflected in the changes to `WorkspaceManagerDAO`. Those functions used to take a raw access token and now take a `RawlsRequestContext` instead so the trace information can be propagated to WSM (and later others).

Also tracing was added to all routes (recommend ignoring whitespace for that). There were some individual routes where tracing had been added which was removed. Plus some auto formatting courtesy of IntelliJ.

[Example trace](https://console.cloud.google.com/traces/list?tid=d11350b38bbae9c0c19af965539c3ee0)

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/WorkspaceManagerApiClientProvider.scala [hunk 1]
```diff
@@ -1,33 +1,61 @@
...
-  def getApiClient(accessToken: String): ApiClient = {
-    val client: ApiClient = new ApiClient()
+  def getApiClient(ctx: RawlsRequestContext): ApiClient = {
+    val client: ApiClient = new ApiClient() {
+      override def performAdditionalClientConfiguration(clientConfig: ClientConfig): Unit = {
+        super.performAdditionalClientConfiguration(clientConfig)
+        ctx.tracingSpan.foreach { span =>
+          clientConfig.register(new WithSpanFilter(span))
+          clientConfig.register(new JerseyTracingFilter(Tracing.getTracer))
+        }
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/WorkspaceManagerApiClientProvider.scala:34]: This sets up the tracing, order is important. See https://github.com/DataBiosphere/terra-common-lib/pull/76 for `JerseyTracingFilter`.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/util/TracingUtils.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import io.opencensus.scala.Tracing
+import io.opencensus.scala.Tracing._
+import io.opencensus.trace.Status
+import org.broadinstitute.dsde.rawls.model.{RawlsRequestContext, RawlsTracingContext}
+import slick.dbio.{DBIO, DBIOAction, Effect, NoStream}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object TracingUtils {
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/util/TracingUtils.scala:11]: This was `OpenCensusDBIOUtils` but was reformulated to be in terms of `RawlsRequestContext` and `RawlsTracingContext`

File: core/src/main/scala/org/broadinstitute/dsde/rawls/model/RawlsRequestContext.scala [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.broadinstitute.dsde.rawls.model
+
+import io.opencensus.trace.Span
+
+final case class RawlsRequestContext(userInfo: UserInfo, tracingSpan: Option[Span] = None)
+final case class RawlsTracingContext(tracingSpan: Option[Span] = None)
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/model/RawlsRequestContext.scala:6]: There is one case where we have some tracing that does not have a user in context.

**Review by davidangb — APPROVED** [review]
Thank you for this! Thumbed modulo needing a rebase/merge

**Review by aherbst-broad — APPROVED** [review]

**Comment by dvoet** [comment]
jenkins retest

**Pull request merged**